src/components/mask-input/mask-input.ts

```diff
@@ -0,0 +1,360 @@
+  }
+
+  protected updateInput(part: string, start: number, finish: number) {
+    const { value, end } = this.parser.replace(
+      this.maskedValue,
+      part,
+      start,
+      finish
+    );
+    this.maskedValue = value;
+    this._value = this.parser.parse(value);
+
+    this.requestUpdate();
+    if (start !== this.mask.length) {
+      this.emitEvent('igcInput', { detail: this.value });
+    }
+    this.updateComplete.then(() => this.input.setSelectionRange(end, end));
+  }
+
+  protected handleCut() {
+    this.selection = this.inputSelection;
+  }
+
+  protected handleDragStart() {
+    this.selection = this.inputSelection;
+  }
+
+  protected handleDragEnter() {
+    if (!this.hasFocus) {
+      this.maskedValue = this.parser.apply(this._value);
+    }
+  }
+
+  protected handleDragLeave() {
+    if (!this.hasFocus) {
+      this.updateMaskedValue();
+    }
+  }
+
+  protected override handleFocus() {
+    this.hasFocus = true;
+    this.maskedValue = this.parser.apply(this._value);
+    super.handleFocus();
+
+    // In case of empty value, select the whole mask
+    if (!this._value) {
+      this.updateComplete.then(() => this.select());
+    }
+  }
+
+  protected override handleBlur() {
+    this.hasFocus = false;
+    this.updateMaskedValue();
+    super.handleBlur();
+  }
+
+  protected override handleChange() {
+    this.emitEvent('igcChange', { detail: this.value });
+    this.invalid = !this.parser.isValidString(this._value);
+  }
+
+  protected handleInvalid() {
+    this.invalid = true;
+  }
+
+  protected handleClick() {
+    // Clicking at the end of the input field will select the entire mask
+    if (
+      this.input.selectionStart === this.input.selectionEnd &&
+      this.input.selectionStart === this.maskedValue.length
+    ) {
+      this.select();
+    }
+  }
+
+  protected updateMaskedValue() {
+    if (this.maskedValue === this.parser.apply()) {
+      this.maskedValue = '';
+    }
+  }
+
+  /** Replaces the selected text in the control and re-applies the mask */
+  public override setRangeText(
+    replacement: string,
+    start: number,
+    end: number,
+    _selectMode: 'select' | 'start' | 'end' | 'preserve' = 'preserve'
+  ) {
+    this.input.value = this.parser.replace(
+      this.input.value,
+      replacement,
+      start,
+      end
+    ).value;
+    this.maskedValue = this.parser.apply(this.parser.parse(this.input.value));
+    this._value = this.parser.parse(this.maskedValue);
+  }
+
+  public override setSelectionRange(
+    start: number,
+    end: number,
+    direction?: 'backward' | 'forward' | 'none'
+  ): void {
+    super.setSelectionRange(start, end, direction);
+    this.selection = { start, end };
+  }
+
+  /** Checks for validity of the control and shows the browser message if it's invalid. */
+  public reportValidity() {
+    return (
+      this.input.reportValidity() && this.parser.isValidString(this._value)
```

Comment:
Same here - only check required positions if there is something filled. Empty value is the responsibility of the `required` attribute rule and will be handled by the native `report/checkValidity`